src/libraries/System.Runtime/src/System.Runtime.csproj

```diff
@@ -3,6 +3,14 @@
     <IsPartialFacadeAssembly>true</IsPartialFacadeAssembly>
     <TargetFramework>$(NetCoreAppCurrent)</TargetFramework>
   </PropertyGroup>
+  <PropertyGroup>
+    <!-- the System.Runtime project is special, we need to be able to hide the System.Private.Uri types for the generated ref .cs so we can include it in the System.Private.CoreLib ref .cs -->
+    <ContractTypesPartiallyMoved>true</ContractTypesPartiallyMoved>
+    <ProjectForGenAPIDocIdGeneration>$(UriProject)</ProjectForGenAPIDocIdGeneration>
+    <GenAPIConditionalTypeListSymbol>!HIDE_URI_TYPES</GenAPIConditionalTypeListSymbol>
```

Comment:
The way it works is that GenAPI will wrap the Uri types (which are implemented in System.Private.Uri.dll) with this conditional compilation symbol defined here so that we can exclude them when compiling the CoreLib reference assembly, e.g. 

```csharp
#if !HIDE_URI_TYPES
public class Uri
{
}
...
#endif
```

Reply:
Why not just use the same define for this that you're using for others, EG: `BUILDING_REFERENCE`.  You could add an item with the typelist to apply it to the URI types.

Curious - does System.Private.Uri need a stable MVID too, or is that less of an issue since we don't cross-compile it as much?

Reply:
@ericstj did that in the latest push. System.Private.Uri will have a stable MVID since it compiles against the CoreLib reference assembly too.